### ISSUE TITLE
Fuel [BLUS30314]: Texture corruption and other graphic issues

### ISSUE THREAD

**Issue by Shoegzer** [RPCS3/rpcs3#10494]
During a race, some of the game's textures appear as white polygons rather than the actual textures.  Also, when in free ride mode after a race, no headlights emit from the vehicle even though the game is dark.  This could either be that time is actually daytime, or that the headlights themselves are not rendered properly.  Playing the game with this issue is therefore difficult given the impaired visibility.

Texture issue (note white polygons on left mountainside):

![shot1](https://user-images.githubusercontent.com/12475371/123482987-12cadd80-d5d4-11eb-8586-e84a385e8d86.png)

Darkness/headlight issue:

![shot2](https://user-images.githubusercontent.com/12475371/123483023-20806300-d5d4-11eb-8792-80331faa6015.png)

Note: For some reason, RPCS3 freezes when creating an RSX capture but can provide one once that issue is resolved.

System details:
* OS: Linux Mint 20.1 / kernel 5.11
* CPU: Intel i7-10870H
* GPU: Nvidia GTX 3070
* Driver version: 460.80
* Backend: Vulkan and OpenGL tested
* Commit tested: bba52857408ec3b7b5df919b4982600dd505466a
* Log: [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/6718804/RPCS3.log.gz)

**Comment by kacek** [comment]
you sure it's not just emulation pausing during capture? check captures folder if anything is there.

**Comment by Shoegzer** [comment]
Thanks @kacek , I don't think it auto-paused the last time I took a capture but it's been a while since I've done one so I could be wrong.  In any event I uploaded a new capture that was active through both issues.

**Comment by benderscruffy** [comment]
does #10764 fix this issue ??

**Comment by Shoegzer** [comment]
Thanks for the heads-up, but unfortunately it doesn't.  I tested soon after #10764 was merged and observed these issues to remain.  My guess is that the main issue (transparent/white polygons in the terrain) are related to the texture cache.  Fuel is by far the largest open-world game on the PS3 and could be doing clever things with memory/cache for its streaming texture data which RPCS3 hasn't accommodated.  Just a guess, I could be way off.

I also observed potentially the same issue in Just Cause 2: at about 1:30 minutes into [this](https://youtu.be/9_np0cUaKJQ?t=89) video you can see the textures on the mountain to the left as transparent blue.  That's pretty much what I see in Fuel - textures appearing incorrectly, then correcting with changes in distance (or camera angle?).  For all I know that issue may have been resolved though, possibly even by the PR you mentioned.

**Comment by kuboslawik** [comment]
As of RPCS3 Version: 0.0.19-13002-ba727e13 Alpha I can confirm that those issues are still present.

**Comment by OmegaRiley** [comment]
Just wanted to confirm this is still an issue on current build (0.0.21-13388-4a8668c Alpha), and provide a bump to the thread. (No clue if that's how any of this works, but probably not)

**Comment by Shoegzer** [comment]
Update: It appears the second issue related to darkness/headlights is fixed!

![pic](https://user-images.githubusercontent.com/12475371/163263206-5d60434b-af60-4454-b668-aee6ab23ca55.png)

Not sure what commit fixed this but I observed the improvement with f6a06a5618a87ab9bf55107a8a8ee5df6b08b186.
 
Unfortunately the main issue related to white/transparent polys is still present, but I'm happy to provide any further information or testing that may help to resolve it.

**Comment by kuboslawik** [comment]
It works all the time for you? For me since 0.19, headlights are occasionally triggered by trucks in free roam, other than that in every version since then, it's broken for me. Additionally, version after 0.19 adds sound crackling.

**Comment by Shoegzer** [comment]
That's too bad, I'll have to re-test for consistency but you may be right.  With my test, I always ran the same race, and when it ended, I opted for "free roam".  This always put me in broad daylight, but within seconds it would suddenly turn to night, though my headlights remained off.  This made me guess that the emulated game was somehow confusing daytime and nighttime (and in that case the headlight issue wouldn't necessarily be rendering-related).  However, in my latest test posted immediately above, daylight remained after the race until the sun went down several minutes later, and I observed the headlights eventually turning on when it became dark enough, as you can see.

I'm not experiencing your audio cracking issue, but I'm guessing that may be an unrelated performance regression that I'm not seeing given my CPU/GPU.  Are you getting full FPS while you hear crackling?

**Comment by kuboslawik** [comment]
FPS is fluctuating between 29.99-30.02, I'm using Linux build. Game might be really confusing day and night, I tested free roam now, lights turned on after after a while in night. But the major problem are challenges, when they are at night, there are no lights :/

**Comment by Shoegzer** [comment]
Interesting, not sure why audio is cracking for you then.  Maybe a linux driver regression?  Re-test with an older RPCS3 build and see.  If it still happens you can then perform a bisect to nail down the problem build.  Regardless, this is getting OT...

What you're saying seems to line up with my experience regarding day/night confusion.  It seems headlights work fine as long as the game is allowed to "transition" to night naturally (allowing sun to set) but if the game is suddenly forced into nighttime mode such as with challenges (which I haven't tested) then headlights don't work.  Does that sound right?  If so, perhaps we should open a separate issue as that doesn't sound rendering-related.  That would allow the main issue with white/transparent textures to be tracked here, independently.

**Comment by kd-11** [comment]
There is a linux audio crackling regression introduced when cubeb was added to the emulator.

**Comment by Shoegzer** [comment]
Thanks.  It seems to be working fine for me though I suppose YMMV.

**Comment by kuboslawik** [comment]
Yes, we should open new issue with day/night cycle. I think the fastest way to test it, is to play 'The Asphalt Medusa' race in tsunami reef camp (first one). Half way through the race there is transition to the night.

Regardinf audio, changing to Cubeb helps, a bit, but occasional crackling is till there.

**Comment by Shoegzer** [comment]
Fair enough.  I opened the issue - please provide the needed details there.  Thanks!

**Comment by rafaelferreiraql** [comment]
I'm also having the missing textures issue, far away textures keep popping on and out of existence.

(I'm on Windows 10, AMD Ryzen 5 5600G with internal graphics, both backends)

**Comment by Shoegzer** [comment]
Thanks @rafaelferreiraql, I was actually hoping the recent #13046 had a shot at fixing this but then I read your post and realized it's not confined to Nvidia. Not sure if that makes this issue easier or harder to fix.  I did find time to test again though, same info as above except for:

* OS: Linux Mint 21
* Driver: 525.60
* Commit: 83995164759e6d21a36ec7c29aec3377cda8cb7f

**Comment by kd-11** [comment]
Still an issue?

**Comment by Shoegzer** [comment]
Unfortunately, yes.  Tested with 6c33c38c08ad6684ed660d10689f45d0b5407963.  I do test this title frequently for any changes.

**Comment by mavethee** [comment]
Seems to still be an issue on RPCS3 v0.0.30-16135
![CleanShot 2024-02-25 at 10 28 54@2x](https://github.com/RPCS3/rpcs3/assets/47083527/63b83a97-35d5-4050-8320-937634081cca)
(Found wildest representation, usually they reappear once I wanna make a screenshot)

Logfile:
[FUEL_ BLES00570.log.gz](https://github.com/RPCS3/rpcs3/files/14395912/FUEL_.BLES00570.log.gz)

This is not really up to region, so title might be changed I guess?

**Comment by Shoegzer** [comment]
Thanks, it helps to know other regions are seeing this too.  I've changed the title though would still like to encourage others to test versions that are media- rather than region-focused (such as the Fuel demo NPUB90281).

**Comment by mavethee** [comment]
> Thanks, it helps to know other regions are seeing this too. I've changed the title though would still like to encourage others to test versions that are media- rather than region-focused (such as the Fuel demo NPUB90281).

@Shoegzer I mean yeah, rarely it would be single region issue, plus I'm on a Mac so additional cursed things happen 💀

**Comment by Shoegzer** [comment]
Update: [This video](https://www.youtube.com/watch?v=DpBBhF0lbYs) clearly shows the texture issue in contrast with proper rendering (although on PC, not PS3).  Look at the mountain range to the left, about 45 seconds in.

**Comment by digant73** [comment]
@Shoegzer is that still an issue? any improvement?

**Comment by Shoegzer** [comment]
No improvement unfortunately.  I always test this title against any changes to graphics/RSX emulation.

**Comment by Shoegzer** [comment]
Tested against build b30a44c1367635af0675f61504e1ca45263655c6 which includes PR #18170.  The texture issue remains though.

**Comment by Shoegzer** [comment]
Tested against PR #18712 with "Enable Special Depth Comparison" on, but no change. Note that this PR primarily addresses Z-Fighting issues (which is unrelated here), but is not entirely limited to them.

**Comment by Shoegzer** [comment]
Tested against PR #19235 but no change.

**Comment by mavethee** [comment]
Testing online features, maintaining high speed and issue is still present.
